python-sdk/nuscenes/scripts/export_kitti.py

```diff
@@ -0,0 +1,347 @@
+            kitti_transforms['P3'] = np.zeros((3, 4))  # Dummy values.
+            kitti_transforms['R0_rect'] = r0_rect.rotation_matrix  # Cameras are already rectified.
+            kitti_transforms['Tr_velo_to_cam'] = np.hstack((velo_to_cam_rot, velo_to_cam_trans.reshape(3, 1)))
+            kitti_transforms['Tr_imu_to_velo'] = imu_to_velo_kitti
+            calib_path = os.path.join(calib_folder, sample_token + '.txt')
+            with open(calib_path, "w") as calib_file:
+                for (key, val) in kitti_transforms.items():
+                    val = val.flatten()
+                    val_str = '%.12e' % val[0]
+                    for v in val[1:]:
+                        val_str += ' %.12e' % v
+                    calib_file.write('%s: %s\n' % (key, val_str))
+
+            # Write label file.
+            label_path = os.path.join(label_folder, sample_token + '.txt')
+            if os.path.exists(label_path):
+                print('Skipping existing file: %s' % label_path)
+                continue
+            else:
+                print('Writing file: %s' % label_path)
+            with open(label_path, "w") as label_file:
+                for sample_annotation_token in sample_annotation_tokens:
+                    sample_annotation = self.nusc.get('sample_annotation', sample_annotation_token)
+
+                    # Get box in LIDAR frame.
+                    _, box_lidar_nusc, _ = self.nusc.get_sample_data(lidar_token, box_vis_level=BoxVisibility.NONE,
+                                                                     selected_anntokens=[sample_annotation_token])
+                    box_lidar_nusc = box_lidar_nusc[0]
+
+                    # Truncated: Set all objects to 0 which means untruncated.
+                    truncated = 0.0
+
+                    # Occluded: Hard-coded: Full visibility.
+                    occluded = 0
+
+                    # Convert nuScenes category to nuScenes detection challenge category.
+                    detection_name = category_to_detection_name(sample_annotation['category_name'])
+
+                    # Skip categories that are not part of the nuScenes detection challenge.
+                    if detection_name is None:
+                        continue
+
+                    # Convert to KITTI 3d and 2d box and KITTI output format.
+                    box_cam_kitti = KittiDB.box_nuscenes_to_kitti(
+                        box_lidar_nusc, Quaternion(matrix=velo_to_cam_rot), velo_to_cam_trans, r0_rect)
+                    box_cam_kitti.score = 0  # Set dummy score so we can use this file as result.
+                    output = KittiDB.box_to_string(name=detection_name, box=box_cam_kitti, truncation=truncated,
+                                                   occlusion=occluded)
+
+                    # Write to disk.
+                    label_file.write(output + '\n')
+
+    def render_kitti(self) -> None:
```

Comment:
I am worried that when we run this in a jupyter notebook it would render the annotations to the notebook output as well. Have we tested this ? If so, we can probably add some description as a warning

Reply:
Tested it on Jupyter Notebook. Works!